model/src/main/kotlin/DependencyGraph.kt

```diff
@@ -0,0 +1,240 @@
+    UNKNOWN
+}
+
+/**
+ * Generate the type of an [Identifier] based on a [dependencyType] and [managerName].
+ */
+private fun identifierType(dependencyType: DependencyType, managerName: String): String =
+    when (dependencyType) {
+        DependencyType.MANAGER -> managerName
+        DependencyType.MAVEN -> "Maven"
+        DependencyType.UNKNOWN -> "Unknown"
+    }
+
+/**
+ * A class to model a tree-like structure to represent the dependencies of a project.
+ *
+ * Instances of this class are used to store the relations between dependencies in fragments of dependency trees in an
+ * Analyzer result. The main purpose of this class is to define an efficient serialization format, which avoids
+ * redundancy as far as possible. Therefore, dependencies are represented by numeric indices into an external table.
+ * As a dependency can occur multiple times in the dependency graph with different transitive dependencies, the class
+ * defines another index to distinguish these cases.
+ *
+ * Note: This is by intention no data class. Equality is tested via references and not via the values contained.
+ */
+// Do not serialize default values to reduce the size of the result file.
+@JsonInclude(JsonInclude.Include.NON_DEFAULT)
+class DependencyReference(
+    /**
+     * Stores the numeric index of the dependency referenced by this object. The package behind this index can be
+     * resolved by evaluating the list of identifiers stored in [DependencyGraph] at this index.
+     */
+    val index: Int,
+
+    /**
+     * Stores the index of the fragment in the dependency graph where the referenced dependency is contained. This is
+     * needed to uniquely identify the target if the dependency occurs multiple times in the graph. Note that this
+     * property and the [index] property correspond to the data stored by a [DependencyIndex] object. Inlining these
+     * properties here rather than embedding a [DependencyIndex] field makes the serialized form more efficient.
+     */
+    val fragmentIndex: Int = 0,
+
+    /**
+     * A set with the references to the dependencies of this dependency. That way a tree-like structure is established.
+     */
+    val dependencies: SortedSet<DependencyReference> = sortedSetOf(),
+
+    /**
+     * The type of linkage used for the referred package from its dependent package. As most of our supported
+     * package managers / languages only support dynamic linking or at least default to it, also use that as the
+     * default value here to not blow up our result files.
+     */
+    @JsonInclude(value = JsonInclude.Include.CUSTOM, valueFilter = PackageLinkageValueFilter::class)
+    val linkage: PackageLinkage = PackageLinkage.DYNAMIC,
+
+    /**
+     * Defines the type of this dependency.
+     */
+    val dependencyType: DependencyType = DependencyType.MANAGER,
+
+    /**
+     * A list of [OrtIssue]s that occurred handling this dependency.
+     */
+    @JsonAlias("errors")
```

Comment:
For this class the field was never named `error`, so I believe we can drop this.